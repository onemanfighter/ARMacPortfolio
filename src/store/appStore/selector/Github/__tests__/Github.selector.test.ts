import { renderHook } from '@testing-library/react-hooks';
import { appStore, githubSelector } from '@appStore';

describe('Github selector', () => {
  it('should return default github state', () => {
    const { result } = renderHook(() => appStore(githubSelector));

    expect(result.current).toMatchSnapshot();
  });

  it('should return current profile', () => {
    const { result } = renderHook(() => appStore(githubSelector));

    expect(result.current.currentProfile).toBe('onemanfighter');
  });

  it('should return default profile', () => {
    const { result } = renderHook(() => appStore(githubSelector));

    expect(result.current.defaultProfile).toBe('onemanfighter');
  });

  it('should return project', () => {
    const { result } = renderHook(() => appStore(githubSelector));

    expect(result.current.getCurrentProject()).toBe(
      'https://onemanfighter.github.io/onemanfighter',
    );
  });

  it('should set project', () => {
    const { result } = renderHook(() => appStore(githubSelector));

    result.current.setProject('ar-mac-portfolio');

    expect(result.current.getCurrentProject()).toBe(
      'https://onemanfighter.github.io/ar-mac-portfolio',
    );
  });

  it('should set current profile', () => {
    const { result } = renderHook(() => appStore(githubSelector));

    result.current.resetCurrentProfile();

    expect(result.current.currentProfile).toBe('onemanfighter');
  });
});
