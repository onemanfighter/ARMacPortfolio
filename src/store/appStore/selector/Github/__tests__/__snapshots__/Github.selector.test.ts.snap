// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github selector should return default github state 1`] = `
{
  "currentProfile": "onemanfighter",
  "defaultProfile": "onemanfighter",
  "getCurrentProject": [Function],
  "resetCurrentProfile": [Function],
  "setCurrentProfile": [Function],
  "setProject": [Function],
}
`;
